cli/src/client/queries.rs

```diff
@@ -0,0 +1,73 @@
+use crate::client::Client;
+use graphql_client::*;
+use std::error::Error;
+
+// We have to declare types for scalars in the schema.
+type GraphQLDocument = String;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    query_path = "graphql/get_schema_query.graphql",
+    schema_path = "graphql/agm.graphql",
+    response_derives = "PartialEq, Debug",
+    deprecated = "warn"
+)]
+pub struct GetSchemaQuery;
+
+pub fn get_schema_query(
+    client: &Client,
+    variables: get_schema_query::Variables,
+) -> Result<Option<GraphQLDocument>, Box<dyn Error>> {
+    let response_body: Response<get_schema_query::ResponseData> =
+        client.send::<GetSchemaQuery>(variables)?;
+
+    let hash: Option<GraphQLDocument> = response_body.data.and_then(|data| {
+        data.service
+            .and_then(|service| service.schema.map(|schema| schema.hash))
+    });
+
+    Ok(hash)
+}
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    query_path = "graphql/me_query.graphql",
+    schema_path = "graphql/agm.graphql",
+    response_derives = "PartialEq, Debug",
+    deprecated = "warn"
+)]
+pub struct MeQuery;
+
+pub fn me_query(client: &Client) -> Result<Option<me_query::MeQueryMe>, Box<dyn Error>> {
+    let response_body: Response<me_query::ResponseData> =
+        client.send::<MeQuery>(me_query::Variables {})?;
+
+    Ok(response_body.data.and_then(|data| data.me))
+}
+
+mod tests {
```

Comment:
Maybe these would be better as [documentation tests](https://doc.rust-lang.org/rustdoc/documentation-tests.html)? Then we get documentation!

Reply:
Looks like we could also [mark them no_run](https://doc.rust-lang.org/rustdoc/documentation-tests.html#attributes)

Reply:
Good suggestion! They really are just there for documentation :)

Reply:
Something is not working as expected with the doctests. I tried moving these over and my first hint that they're not working is that IntelliJ doesn't render them as rust code, but as simple comments. So I tried adding a compilation error in them, and everything still passes.
I went to read some docs and couldn't for the life of me figure out how to do this.
I took this example 

```
/// First line is a short summary describing function.
///
/// The next lines present detailed documentation. Code blocks start with
/// triple backquotes and have implicit `fn main()` inside
/// and `extern crate <cratename>`. Assume we're testing `doccomments` crate:
///
/// ```
/// let result = doccomments::add(2, 3);
/// assert_eq!(res, 5);
/// ```
pub fn add(a: i32, b: i32) -> i32 {
    a + b
}
```
and changed the last line from `result` to `res` in the assert to make it not compile, and everything still passes. I tried putting this block in `main.rs` too just to make sure.

I then tried putting that `add` block in a different crate (something I set up locally, the minigrep project from The Book) and IT RENDERS THERE. I seriously am lost on this.

Reply:
Wow.. Figured this out. So I tried running this:
```
$ cargo test --doc
```
And got the response:
```
error: no library targets found in package `apollo-cli`
```

So, we have as `main.rs` but no `lib.rs` -- I created that file and moved all of the `mod ___;` from `main.rs` to `lib.rs` and NOW DOCTESTS WORK.

🤦

Reply:
https://github.com/rust-lang/rust/issues/50784

Reply:
Thinking about this, I'd rather not introduce a `lib.rs` just yet for the sole purpose of doctests. It forces us to mark some modules as public just that `main` could use them. So, I think I'll keep the code as is so that tests will compile.